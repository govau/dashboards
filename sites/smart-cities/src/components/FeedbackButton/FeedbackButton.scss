@import '~style/main';

.button {
  background: $color-primary--800;
  height: 32px;
  border-radius: 20px;
  font-size: 14px;
  text-align: right;
  color: $color-white;

  &.pulsing {
    animation: emit 1s ease 1s 3;
  }

  &.dimmed {
    background: $color-grey--700;
  }

  &:hover {
    background: $color-primary--900;
  }

  @include for-desktop-up {
    width: 176px;
    padding-top: 5px;
  }
}

.buttonText {
  display: none;

  @include for-desktop-up {
    display: inline;
  }
}

.speechBubbleIcon {
  margin: 4px 16px;

  @include for-desktop-up {
    margin-left: 18px;
    margin-top: -2px;
    margin-right: 8px;
  }
}

@keyframes emit {
  from { box-shadow: 0 0 0 0 rgba(255, 255, 255, 0.6); }
  to { box-shadow: 0 0 0 15px rgba(255, 255, 255, 0); }
}
