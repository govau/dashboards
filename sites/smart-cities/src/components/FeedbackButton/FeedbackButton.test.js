import React from 'react';
import { shallow } from 'enzyme';
import FeedbackButton from './FeedbackButton';

it('should match snapshot', () => {
  const component = shallow(
    <FeedbackButton />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should dim when required', () => {
  const component = shallow(
    <FeedbackButton dimmed />
  );

  const button = component.find('.button');
  expect(button.hasClass('dimmed')).toBe(true);
});

it('should not dim when not required', () => {
  const component = shallow(
    <FeedbackButton />
  );

  const button = component.find('.button');
  expect(button.hasClass('dimmed')).toBe(false);
});
