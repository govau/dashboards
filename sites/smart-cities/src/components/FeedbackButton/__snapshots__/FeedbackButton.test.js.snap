// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
"<NavLink to=\\"/feedback\\" className=\\"button\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
  <span className=\\"buttonText\\">
    Give feedback
  </span>
  <Icon icon=\\"speechBubble\\" className=\\"speechBubbleIcon\\" size={25} />
</NavLink>"
`;
